app/views/devise/registrations/new.html.erb

```diff
@@ -0,0 +1,40 @@
+<div class="user">
+
+<h2>Sign up</h2>
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+  <div class="field">
+    <%= f.label :name %><br />
+    <%= f.text_field :name, autofocus: true, autocomplete: "name" %>
+  </div>
+
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+    <em>(<%= @minimum_password_length %> characters minimum)</em>
+    <% end %><br />
+    <%= f.password_field :password, autocomplete: "new-password" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+    <%= f.hidden_field :posts_counter, value: 0 %>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
```

Comment:
:red_circle: [REQUIRED] I'm reviewing your work on the confirmation email feature. I noticed that you haven't implemented the feature yet.
That's okay! [The project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md) only say that you need to **ask for confirmation by email**. You don't actually have to send a confirmation email.

To test the feature, you can manually update the confirmed_at column in the users table. This will allow the user to access the root page.

Once you're happy with the feature, you can implement the confirmation email. However, this is not mandatory.

The other thing I noticed is that any new user can access the root page after signing up. This is not correct as requested by the previous reviewer :arrow_down: . The correct flow is to block any user from accessing the root page unless the account is verified.

> * [ ]  Good job adding `:confirmable` to your User model authentication. However, as required for the for the project, you should ensure that User cannot be able to login before his `email` has been confirmed, because it is not the same in your case. Right now, after a user has singed up, he is immediately logged in even if without being confirmed. Make sure you fix that.
>       ![Screenshot from 2023-08-03 10-45-10](https://user-images.githubusercontent.com/83042742/258055631-e0783b13-5fc2-4e50-806f-60ecd2114ed4.png)



You can fix this by adding a check in the ApplicationController to see if the user is confirmed. If the user is not confirmed, you can redirect them to the confirmation page.

I hope this helps! Let me know if you have any other questions.

Reply:
Hello @bienvenuushindi I used a gem called 'letter opener' which confirms the user before navigating to the users#index page. I have tried to implement it using the image, but it is not confirming. I have updated the ApplicationController but not working yet. 

The project requirement does not specify we use any given method so using an extra gem 'letter opener' should be a problem.

This is the ApplicationController

` class ApplicationController < ActionController::Base
  protect_from_forgery with: :exception
  before_action :authenticate_user!

  before_action :update_allowed_parameters, if: :devise_controller?

  protected

  def update_allowed_parameters
    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :email, :password, :posts_counter) }
    devise_parameter_sanitizer.permit(:account_update) { |u| u.permit(:name, :email, :password, :current_password) }
  end

end `

Reply:
Hello @prius2055 

I understand that you're using the letter opener gem to confirm users before they can navigate to the users#index page. However, it seems that the gem is not working as expected.
If the letter opener gem was working correctly, the confirmed user would be able to log in. However, this is not the case currently.

As I mentioned in my previous email, it is not mandatory to implement the confirmation email feature in this milestone. If you're having trouble getting the letter opener gem to work, you can just assume that this feature is a future feature.

Any person who wants to test the app should update the confirmed_at column in the database for testing purposes. This will allow them to log in even if they have not confirmed their email address.

Reply:
Hi @bienvenuushindi, the 'letter opener' is working perfectly from my end.

These are  the screenshot. 
![gem](https://github.com/prius2055/rails-blogging-app/assets/63577080/b5746efd-9085-4ecf-bd02-b65221404332)

![gem2](https://github.com/prius2055/rails-blogging-app/assets/63577080/b8e53afd-cee2-4c50-9fba-9bea248f6075)

Please check again from your end

Reply:
Fixed